lib/scout_apm/tracked_request.ex

```diff
@@ -48,6 +48,16 @@ defmodule ScoutApm.TrackedRequest do
     stop_layer(tr, fn x -> x end)
   end
 
+  def stop_layer(%__MODULE__{layers: []} = tracked_request, callback) when is_function(callback) do
+    ScoutApm.Logger.warn("Scout Layer mismatch when stopping layer in #{inspect(tracked_request)}")
+    :error
+  end
+
+  def stop_layer(%__MODULE__{children: []} = tracked_request, callback) when is_function(callback) do
+    ScoutApm.Logger.warn("Scout Layer mismatch when stopping layer in #{inspect(tracked_request)}")
+    :error
+  end
```

Comment:
I worry about any `warn` level logs that could trigger on every single request. We've had this a few times in the Ruby agent, where we grew logs to multiple gigabytes on our customer's systems.  Lower the error to info or debug to prevent this.

I know it feels like an error or warn level log, but if this is happening, it's not actionable by the user, so I want to be sure we don't blow up a user's log dir in that case. What'll happen is that they contact us, and we'll have them turn up logging to figure out what's going on, and fix the underlying bug causing it.